repo/packages/L/linkerd/2/marathon.json.mustache

```diff
@@ -0,0 +1,54 @@
+{
+  "id": "{{linkerd.name}}",
+  "instances": {{linkerd.instances}},
+  "cpus": {{linkerd.cpus}},
+  "mem": {{linkerd.mem}},
+  "acceptedResourceRoles": ["*", "slave_public"],
+  "constraints": [["hostname", "UNIQUE"]],
+  "maintainer": "hello@buoyant.io",
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "{{resource.assets.container.docker.linkerd-docker}}",
+      "network": "BRIDGE",
+      "privileged": true,
+      "portMappings": [
+        {
+          "containerPort": {{linkerd.admin-port}},
+          "hostPort": {{linkerd.admin-port}},
+          "servicePort": 0,
+          "protocol": "tcp"
+        },
+        {
+          "containerPort": {{linkerd.routing-port}},
+          "hostPort": {{linkerd.routing-port}},
+          "servicePort": 0,
+          "protocol": "tcp"
+        }
+      ]
+    }
+  },
+  "healthChecks": [
+    {
+      "protocol": "HTTP",
+      "portIndex": 0,
+      "path": "/admin/ping"
+    }
+  ],
+  "args": [
+    "/mnt/mesos/sandbox/{{linkerd.config-filename}}"
+  ],
+  "requirePorts":true,
+  "ports": [
+    {{linkerd.admin-port}},
+    {{linkerd.routing-port}}
+  ],
+  "uris": [
+    "{{linkerd.config-uri}}/{{linkerd.config-filename}}"
```

Comment:
In general user configurable value should be passed to the task by either command line flags or environment variables. What some packages do is grab those environment variable and command line flags and create the configuration files before launching the long lived process.

The general issue with configuration through files/uri is that they need to get hosted so that the Mesos Agent can download them through HTTP. Universe and Cosmos currently don't support this functionality so it complicate deployment for our users.